### ISSUE TITLE
ETM: Apical cell numbers are unstable

### ISSUE THREAD

**Issue by mrcslws** [numenta/nupic.research#565]
The `ExtendedTemporalMemory` (formerly known as `GeneralTemporalMemory`) refers to "external" and "apical" cells by shifting their indices. For example, if there are 1024 cells in the TM, then the 1025th cell is the first "external" cell. These indices are used for synapses -- cells within the TM connect to cell indices.

The "apical" cells are similarly placed after the "external" cells. Or, at least, that's the intention.

Here's the actual line: https://github.com/numenta/nupic.research/blob/9cd18ad77f8c5f264c0d41542feadb51b0b70385/htmresearch/algorithms/extended_temporal_memory.py#L91

With this approach, it should be using the number of external cells as the offset. Instead, it's using the number of **active** external cells. This means that if the number of active external cells changes, the indices for the active apical cells will get mangled.

This will mess up any experiments that have a varying active external cell count. It will cause lots of unexpected synapse behavior.

(This issue is independent of any recent changes.)

**Comment by mrcslws** [comment]
The fix is actually really easy: we don't need to offset apical cell indices. They live in their own universe, since there's a Connections instance devoted to just apical connections.

It's helpful to think of each Connections instance as having its own "index space", similar to how processes on a computer have an independent "address space".

It's probably still best to offset them by the number of cells in the TM, since the Connections class is aware of postsynaptic cells, and we shouldn't have "42" mean something different for presynaptic vs postsynaptic. So the apical and basal cell indices will collide, and that will be fine.

In other words, my suggested fix is simply to stop considering the size of the external cells input. We could use today's `_reindexActiveExternalCells` method to handle apical cells.
